modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/LocalWalModeChangeDuringRebalancingSelfTest.java

```diff
@@ -234,16 +230,6 @@ public void testWalDisabledDuringRebalancing() throws Exception {
         doTestSimple();
     }
 
-    /**
```

Comment:
Why so many tests are removed? It isn't related with NPE in `GridMetricManager`.

Reply:
The changes in the tests doesn't relate to the fix.
I removed it to run test that reproduces error many times on the TC - https://ci.ignite.apache.org/viewLog.html?buildId=4969452&.

To fix the issue only changes in GridMetricManager required.

If you know the way write the reproducer, please, let me know.

Reply:
I believe it's impossible to write adequate reproducer except of a lot of test repetition. But I think that configuration should not be deleted on cache stop, only on cache destroy.